### ISSUE TITLE
idea: plugins?

### ISSUE THREAD

**Issue by yoshuawuyts** [stackcss/sheetify#115]
I want to write a package to dedupe CSS; all of it. Problem is: we can't access the whole bundle because we stream it out. Should we perhaps introduce a plugin system of the sorts and have a labeled pipeline thing like browserify does? Perhaps only allow access to the final output stream?

cc/ @ahdinosaur @hughsk thoughts on how we can apply transformations to _all_ of the CSS? Thanks!

**Comment by yoshuawuyts** [comment]
This was brought up by https://github.com/datproject/datfolder/pull/381, which shows we need to dedupe - best would be to operate on _all_ of the CSS ghmm

**Comment by ahdinosaur** [comment]
the simplest option feels like a "global plugin": a function to create a transform stream that is piped the final output stream, but happy with whatever you think is the best way to support your use case. :cherry_blossom:

**Comment by yoshuawuyts** [comment]
Yeah, think so too - tho in on the other hand Browserify is super mature
and their plugin design is well thought out; perhaps we should make the
leap and copy it? idk
